packages/block-library/src/post-excerpt/index.php

```diff
@@ -70,3 +76,21 @@ function register_block_core_post_excerpt() {
 	);
 }
 add_action( 'init', 'register_block_core_post_excerpt' );
+
+/**
+ * If themes or plugins filter the excerpt_length, we need to
+ * override the filter in the editor, otherwise
+ * the excerpt length block setting has no effect.
+ * Returns 100 because 100 is the max length in the setting.
+ */
+if ( is_admin() ||
+	defined( 'REST_REQUEST' ) ||
+	'REST_REQUEST' ) {
```

Comment:
This is very wide net to cast which will affect all instances of excerpt filtering, regardless of source (Post Excerpt block, Latest Post block, traditional theme templates, etc.) — see #48403.

Why not surround the rendering logic in `render_block_core_post_excerpt` with a pair of `add_filter` / `remove_filter` calls?

Reply:
What I meant by my suggestion was something like this (untested):

```php
function render_block_core_post_excerpt( $attributes, $content, $block ) {
    global $_block_core_post_excerpt_length;

    // ...

    $_block_core_post_excerpt_length = int( $attributes['excerptLength'] );
    add_filter( 'excerpt_length', '_block_core_post_excerpt_filter', PHP_INT_MAX );
    get_the_excerpt();
    remove_filter( 'excerpt_length', '_block_core_post_excerpt_filter' );

    // ...
```

with

```php
function _block_core_post_excerpt_filter() {
  global $_block_core_post_excerpt_length;
  return $_block_core_post_excerpt_length;
}
```

The whole `global` dance is a possible way to work with the fact that `$attributes['excerptLength']` is dynamic.

Reply:
There is already a pull request open that tries to address this.

Reply:
> There is already a pull request open that tries to address this.

Which?

Reply:
> > There is already a pull request open that tries to address this.
> 
> Which?

https://github.com/WordPress/gutenberg/pull/48598

Reply:
Not sure if it's the one @carolinan was referring to, but I opened #48598 when I discovered that my excerpts outside of the block editor were not obeying my own excerpt_length filter and traced it back to the line you referenced here:

> This is very wide net to cast which will affect all instances of excerpt filtering, regardless of source (Post Excerpt block, Latest Post block, traditional theme templates, etc.) — see #48403.

My [last revision](https://github.com/WordPress/gutenberg/pull/48598/commits/dc82d46dc137b337eaa9433a2c8ff9f3fe7758f3) in my PR did something like what you suggest here:

> Why not surround the rendering logic in `render_block_core_post_excerpt` with a pair of `add_filter` / `remove_filter` calls?

But it only applies to the rendered block on the page; it is not reflected in the editor. The condition on `REST_REQUEST` is the part that makes it work in the editor. I was also concerned that condition casts too wide a net and raised the question in my PR, but received no feedback on that point.

Reply:
Yes, that is the PR I was thinking of